src/DynamoCore/Models/NodeModel.cs

```diff
@@ -191,6 +191,54 @@ public bool IsUpstreamVisible
         }
 
         /// <summary>
+        /// Input nodes are used in Customizer and Presets. Input nodes can be numbers, number sliders,
+        /// strings, bool, code blocks and custom nodes, which don't specify path.
+        /// </summary>
+        public bool IsInputNode
+        {
+            get
+            {
+                return !inPorts.Any() && !(this is DSFunction);
+            }
+        }
+
+        /// <summary>
+        /// Some of input nodes are not allowed in customizer. Such as directory path or file path.
+        /// If node is not allowed in customizer, this property should be overridden.
+        /// </summary>
+        public virtual bool IsAllowedInCustomizer
```

Comment:
Let's omit this field.  Instead, DynamoPublish should have a dictionary of allowed `NodeModel` types.  It should prevent submission if the node is of a type that is not allowed.